plugins/image-snapshot/src/main.ts

```diff
@@ -0,0 +1,160 @@
+import Proof, { ProofPlugin, ProofTest, TestHookArgs } from '@proof-ui/core';
+import { Browser } from '@proof-ui/browser';
+import CLIPlugin, { CLIOption } from '@proof-ui/cli-plugin';
+import { SnapshotState } from 'jest-snapshot';
+import { toMatchImageSnapshot, MatchImageSnapshotOptions } from 'jest-image-snapshot';
+import kebabCase from 'lodash/kebabCase';
+import traverse from '@babel/traverse';
+import sharp from 'sharp';
+import fs from 'fs';
+
+const SCREEN_SHOT_DIR = '.proof-image-snapshot-temp';
```

Comment:
It may be worth using `os.tmpdir()` or [tmp](https://www.npmjs.com/package/tmp) to create a temp directory specific to each proof run to avoid collisions in the event multiple runs are going at the same time, then cleaning it up at the end of the test.